File: backend/src/main/java/org/persistencia/projetovacine/model/Vacina.java
package org.persistencia.projetovacine.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Vacina {
    public static Integer PERIODICIDADE_DIA = 1;
    public static Integer PERIODICIDADE_SEMANA = 2;
    public static Integer PERIODICIDADE_MES = 3;
    public static Integer PERIODICIDADE_ANO = 4;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String titulo;
    private String descricao;
    private int doses;
    private Integer periodicidade;
    private int intervalo;
}

File: frontend/src/app/interface/vacina.ts
export interface Vacina {
  id?: number
  titulo?: string
  descricao?: string
  doses?: number
  periodicidade?: number
  intervalo?: number
}

export enum Periodicidade {
  DIA = 1,
  SEMANA = 2,
  MES = 3,
  ANO = 4,
}
